lib/active_merchant/billing/gateways/reach.rb

```diff
@@ -0,0 +1,176 @@
+      end
+
+      def purchase(money, payment, options = {})
+        options[:capture] = true
+        authorize(money, payment, options)
+      end
+
+      def capture(money, authorization, options = {})
+        merchant_id, order_id = authorization.split '#'
+        post = { request: { MerchantId: merchant_id, OrderId: order_id } }
+
+        commit('capture', post)
+      end
+
+      def supports_scrubbing?
+        true
+      end
+
+      def scrub(transcript)
+        transcript
+      end
+
+      private
+
+      def build_checkout_request(amount, payment, options)
+        {
+          MerchantId: @options[:merchant_id],
+          ReferenceId: options[:order_id],
+          ConsumerCurrency: options[:currency] || currency(options[:amount]),
+          Capture: options[:capture] || false,
+          PaymentMethod: PAYMENT_METHOD_MAP[payment.brand.to_sym],
+          Items: [
+            Sku: SecureRandom.alphanumeric,
+            ConsumerPrice: amount,
+            Quantity: 1
+          ],
+          ViaAgent: true # Indicates this is server to server API call
+        }
+      end
+
+      def add_payment(payment)
+        {
+          Name: payment.name,
+          Number: payment.number,
+          Expiry: { Month: payment.month, Year: payment.year },
+          VerificationCode: payment.verification_value
+        }
+      end
+
+      def add_customer_data(request, options, payment)
+        address = options[:billing_address] || options[:address]
+
+        return if address.blank?
+
+        request[:Consumer] = {
+          Name: payment.respond_to?(:name) ? payment.name : address[:name],
+          Email: options[:email],
+          Address: address[:address1],
+          City: address[:city],
+          Country: address[:country]
+        }.compact
+      end
+
+      def sign_body(body)
+        Base64.strict_encode64(OpenSSL::HMAC.digest('sha256', @options[:secret].encode('utf-8'), body.encode('utf-8')))
+      end
+
+      def parse(body)
+        hash_response = URI.decode_www_form(body).to_h.transform_keys!(&:to_sym)
+        hash_response[:response] = JSON.parse(hash_response[:response], symbolize_names: true)
+
+        hash_response
+      end
+
+      def format_and_sign(post)
+        post[:request] = post[:request].to_json
+        post[:card] = post[:card].to_json if post[:card].present?
+        post[:signature] = sign_body(post[:request])
+        post
+      end
+
+      def commit(action, parameters)
+        body = post_data format_and_sign(parameters)
+        raw_response = ssl_post url(action), body
+        response = parse(raw_response)
+
+        Response.new(
+          success_from(response),
+          message_from(response),
+          response,
+          authorization: authorization_from(response[:response]),
+          # avs_result: AVSResult.new(code: response['some_avs_response_key']),
+          # cvv_result: CVVResult.new(response['some_cvv_response_key']),
```

Comment:
If we know they do not support these, we can remove the comments.

Reply:
@curiousepic I wanted to leaved there as reminder that there is a pending question for Reach to enable if possible that feature.